libraries/AP_Mount/AP_Mount_Tusuav.cpp

```diff
@@ -0,0 +1,1109 @@
+            mnt_target.target_type = MountTargetType::ANGLE;
+        }
+        break;
+
+    // point mount to another vehicle
+    case MAV_MOUNT_MODE_SYSID_TARGET:
+        if (get_angle_target_to_sysid(mnt_target.angle_rad)) {
+            mnt_target.target_type = MountTargetType::ANGLE;
+        }
+        break;
+
+    default:
+        // we do not know this mode so raise internal error
+        INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
+        break;
+    }
+
+    // send target angles or rates depending on the target type
+    MessageGimbalAngleRates gimbal_angle_mode = {0};
+    switch (mnt_target.target_type) {
+    case MountTargetType::ANGLE:
+        if (mnt_target.angle_rad.yaw_is_ef) {
+            gimbal_angle_mode.mode.bits.yaw = static_cast<uint8_t>(GimbalAngleMode::Earth_Frame_Angle);
+            debug("euler：%f  %f", gimbal_angle_mode.ef_angle[0], gimbal_angle_mode.ef_angle[2]);
+        } else {
+            gimbal_angle_mode.mode.bits.yaw = static_cast<uint8_t>(GimbalAngleMode::Body_Frame_Angle);
+            debug("enc：%f  %f", gimbal_angle_mode.bf_angle[0], gimbal_angle_mode.bf_angle[2]);
+        }
+        gimbal_angle_mode.mode.bits.pitch = static_cast<uint8_t>(GimbalAngleMode::Earth_Frame_Angle);
+        gimbal_angle_mode.mode.bits.roll = static_cast<uint8_t>(GimbalAngleMode::Stabilize);
+        gimbal_angle_mode.mode.bits.special = 0;
+        gimbal_angle_mode.ef_angle[0] = degrees(mnt_target.angle_rad.pitch);
+        gimbal_angle_mode.ef_angle[2] = degrees(mnt_target.angle_rad.yaw);
+        debug("ANGLE_mode：%d", (uint8_t)gimbal_angle_mode.mode.all);
+        break;
+    case MountTargetType::RATE:
+        gimbal_angle_mode.mode.bits.pitch = static_cast<uint8_t>(GimbalAngleMode::Stabilize);
+        gimbal_angle_mode.mode.bits.yaw = static_cast<uint8_t>(GimbalAngleMode::Stabilize);
+        debug("ANGLE_mode：%d", (uint8_t)gimbal_angle_mode.mode.all);
+        break;
+    }
+    send_target_angles_rates(gimbal_angle_mode);
+}
+
+// return true if healthy
+bool AP_Mount_Tusuav::healthy() const
+{
+    // unhealthy until gimbal has been found and replied with firmware version info
+    if (!_initialised) {
+        return false;
+    }
+    // unhealthy if attitude information NOT received recently
+    const uint32_t now_ms = AP_HAL::millis();
+    if (now_ms - _last_current_angle_rad_ms > AP_MOUNT_TUSUAV_HEALTH_TIMEOUT_MS) {
+        return false;
+    }
+    return true;
+}
+
+// get attitude as a quaternion.  returns true on success
+bool AP_Mount_Tusuav::get_attitude_quaternion(Quaternion& att_quat)
+{
+    att_quat.from_euler(_current_angle_rad.x, _current_angle_rad.y, _current_angle_rad.z);
+    return true;
+}
+
+// reading incoming packets from gimbal and confirm they are of the correct format
+void AP_Mount_Tusuav::read_incoming_packets()
+{
+    // check for bytes on the serial port
+    int16_t nbytes = MIN(_uart->available(), 1024U);
```

Comment:
```suggestion
    const auto nbytes = MIN(_uart->available(), 1024U);
```

available actually returns unsigned

Reply:
Thank you for the observation. We have implemented the suggested change to use const auto.